dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+     * Should: return 200 Http code
+     *
+     * @throws ServletException
+     * @throws IOException
+     */
+    @Test
+    public void whenRequestVanityURL() throws ServletException, IOException, DotDataException, DotSecurityException {
+
+        final VelocityServlet velocityServlet = new VelocityServlet();
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template).nextPersisted();
+        ContentletDataGen.publish(htmlPageAsset);
+
+        when(request.getRequestURI()).thenReturn("/vanityURL");
+
+        final String VANITY_URI = "/vanityURL";
+        createAndPublishVanityURL(htmlPageAsset.getPageUrl(), VANITY_URI);
+
+        when(request.getRequestURI()).thenReturn(VANITY_URI);
+        final FilterChain chain = mock(FilterChain.class);
+
+        final VanityURLFilter vanityURLFilter = new VanityURLFilter();
+        vanityURLFilter.doFilter(request, response, chain);
+
+        velocityServlet.service(request, response);
+
+        verify(servletOutputStream).write("".getBytes());
+        verify(response, never()).sendError(HttpServletResponse.SC_NOT_FOUND);
+    }
+
+    private void createAndPublishVanityURL(final String forwardURL, final String VANITY_URI)
+            throws DotDataException, DotSecurityException {
+        final Language defaultLanguage = APILocator.getLanguageAPI().getDefaultLanguage();
+        final Contentlet vanityUrl = FiltersUtil.getInstance().createVanityUrl("test", host.getIdentifier(), VANITY_URI,
+                forwardURL, 200, 0, defaultLanguage.getId());
+
+        FiltersUtil.getInstance().publishVanityUrl(vanityUrl);
+    }
+
+    /**
+     * Method to test: {@link VelocityServlet#service(HttpServletRequest, HttpServletResponse)}
+     * when: There is exists a VanityURL that forward to a URL Map
+     * Should: return 200 Http code
+     *
+     * @throws ServletException
+     * @throws IOException
+     */
+    @Test
+    public void whenRequestURLMapAndVanityURLTogether() throws ServletException, IOException, DotSecurityException, DotDataException {
+
+        final VelocityServlet velocityServlet = new VelocityServlet();
+
+        final String newsPatternPrefix =
+                TEST_PATTERN + System.currentTimeMillis() + "/";
+        final Contentlet contentlet = createURLMapperContentType(newsPatternPrefix, host);
+        ContentletDataGen.publish(contentlet);
+
+        when(request.getRequestURI()).thenReturn("/vanityURL/" + contentlet.getStringProperty("urlTitle"));
+
+        final String VANITY_URI = "/vanityURL/([a-zA-Z0-9-_]+)";
+        final String FORWARD_URL = newsPatternPrefix + "$1";
+
+        createAndPublishVanityURL(FORWARD_URL, VANITY_URI);
+
+        when(request.getRequestURI()).thenReturn("/vanityURL/" + contentlet.getStringProperty("urlTitle"));
+        final FilterChain chain = Mockito.mock(FilterChain.class);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.mock' due to existing static import 'org.mockito.Mockito.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4842460)